persistence-elasticsearch/core/src/main/java/org/apache/unomi/persistence/elasticsearch/ElasticSearchPersistenceServiceImpl.java

```diff
@@ -2081,13 +2081,25 @@ public T executeInClassLoader(Object... args) throws Exception {
         public T catchingExecuteInClassLoader(boolean logError, Object... args) {
             try {
                 return executeInClassLoader(timerName, args);
-            } catch (Throwable t) {
-                if (logError) {
-                    logger.error("Error while executing in class loader", t);
+            } catch (IllegalStateException e) {
+                // Reactor stopped - recovery is not possible
+                if (e.getMessage().contains("I/O reactor status: STOPPED")) {
+                    logger.error("I/O Reactor stopped - stopping application");
+                    System.exit(-1);
```

Comment:
System.exit is really not recommended in an OSGi environment. It would be better to use the UnomiManagementService.stopUnomi method.

Ideally we would add a way for the ElasticSearch persistence implementation to recover from a failing connection but that's a lot more involved in terms of work.

Reply:
Thank you for the code review. I'll fix that soon